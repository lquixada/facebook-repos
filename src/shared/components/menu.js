import React from 'react';
import styled from 'styled-components';

import {Link} from './link';

export const Menu = ({repos}) => (
  <Wrapper>
    <List>
      {repos.map((repo) => (
        <Item key={repo.name}>
          <Link to={repo} />
        </Item>
      ))}
    </List>
  </Wrapper>
);

const Wrapper = styled.div`
  background-color: #fff;
  border: .1rem solid #e5e5e5;
  border-radius: .6rem;
  box-shadow: 0 .1rem .1rem rgba(0,0,0,.1);
  margin-bottom: 1rem;
  font-size: 1.2rem;
  padding: 0;
`;

const List = styled.ul`
  border-radius: .6rem;
`;

const Item = styled.li`
  list-style-type: none;

  &:first-child {
    border-radius: .6rem .6rem 0 0;
    overflow: hidden;
  }

  &:last-child {
    border-radius: 0 0 .6rem .6rem;
    overflow: hidden;
  }
`;

