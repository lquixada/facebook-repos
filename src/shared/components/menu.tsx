import React from 'react'
import styled from 'styled-components'

import {phones} from '../helpers'
import {Count} from '../types'
import {Link} from './link'

interface IProps {
  owner: string
  items: Count[]
}

export const Menu: React.SFC<IProps> = ({owner, items}) => (
  <Wrapper>
    <List>
      {items.map((item) => (
        <Item key={item[0]}>
          <Link
            owner={owner}
            repoName={item[0]}
            contributorsCount={item[1]}
            isLoading={item[2]}
          />
        </Item>
      ))}
    </List>
  </Wrapper>
) as React.ReactElement<any>

const Wrapper = styled.div`
  margin-bottom: 1rem;
  padding: 0;
  max-height: 40rem;
  color: #333;
  font-size: 1.2rem;
  background-color: #fff;
  border: .1rem solid #e5e5e5;
  border-radius: .6rem;
  box-shadow: 0 .1rem .1rem rgba(0,0,0,.1);
  overflow-y: auto;

  @media (${phones}) {
    max-height: 20rem;
  }
`

const List = styled.ul`
  border-radius: .6rem;
`

const Item = styled.li`
  &:first-child {
    border-radius: .6rem .6rem 0 0;
    overflow: hidden;
  }

  &:last-child {
    border-radius: 0 0 .6rem .6rem;
    overflow: hidden;
  }
`
