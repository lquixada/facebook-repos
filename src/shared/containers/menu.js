import {connect} from 'react-redux';
import {withRouter} from 'react-router';
import {List} from 'immutable';

import {Menu} from '../components/menu';

const mapStateToProps = ({reposContributorsCount, repo}, {match}) => ({
  reposContributorsCount: reposContributorsCount.map((item) => {
    const repoName = item.get(0);
    const count = item.get(1);
    const isLoading = repo.getIn([repoName, 'isLoading']);

    return List.of(repoName, count, isLoading);
  })
});
// We need "withRouter" in order to NavLink active state work
export default withRouter(connect(mapStateToProps)(Menu));
