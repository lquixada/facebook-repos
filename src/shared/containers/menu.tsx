import get from 'lodash.get'
import React from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'

import * as actions from '../actions'
import {Menu} from '../components/menu'
import {Count, ICounts} from '../types'

interface IProps {
  counts: ICounts
  items: Count[]
  owner: string
}

export class MenuContainer extends React.Component<IProps, any> {
  public hasLoaded() {
    const data = get(this.props.counts, `${this.props.owner}.data`)
    return !!data
  }

  public render() {
    if (!this.hasLoaded()) {
      return <div style={{color: '#333'}}>Loading...</div>
    }

    return (
      <Menu owner={this.props.owner} items={this.props.items} />
    )
  }
}

const mapStateToProps = ({counts, repo}, {match}) => ({
  counts,
  items: get(counts, `${match.params.owner}.data`, []).map((item) => {
    const {owner} = match.params
    const repoName = item[0]
    const count = item[1]
    const isLoading = get(repo, `${owner}.${repoName}.isLoading`)

    return [repoName, count, isLoading]
  }),
  owner: match.params.owner,
})

// We need "withRouter" in order to NavLink active state work
export default withRouter(connect(mapStateToProps, actions)(MenuContainer))
