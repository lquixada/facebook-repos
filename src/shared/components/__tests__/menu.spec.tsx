import {shallow} from 'enzyme'
import React from 'react'

import {Menu} from '../menu'

describe('<Menu />', () => {
  it('matches the snapshot', () => {
    const items: Array<[string, number, boolean]> = [
      ['repo1', 2, false],
    ['repo2', 5, false],
    ]
    const component = shallow(<Menu owner='some-owner' items={items} />)
    expect(component).toMatchSnapshot()
  })
})
