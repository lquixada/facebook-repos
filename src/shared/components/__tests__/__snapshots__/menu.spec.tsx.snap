// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches the snapshot 1`] = `
<styled.div>
  <styled.ul>
    <styled.li
      key="repo1"
    >
      <Link
        contributorsCount={2}
        isLoading={false}
        owner="some-owner"
        repoName="repo1"
      />
    </styled.li>
    <styled.li
      key="repo2"
    >
      <Link
        contributorsCount={5}
        isLoading={false}
        owner="some-owner"
        repoName="repo2"
      />
    </styled.li>
  </styled.ul>
</styled.div>
`;
